### ISSUE TITLE
Write wrapper for serialize and write to file, as well as load from file and deserialize

### ISSUE THREAD

**Issue by dmcconachie** [UM-ARM-Lab/arc_utilities#25]
The serialize wrapper would presumably take an object, a filename, and a function pointer to an object serialization function. It would serialize, compress, and then save. Something along these lines:

```
template<typename T, typename SerializationFn>
void WriteToFile(const std::string& filename, const T& obj, const SerializationFn& serializer)
{
    std::vector<uint8_t> buffer;
    serializer(obj, buffer);
    ...
}
```

**Comment by dmcconachie** [comment]
This would essentially replace some of the code from https://github.com/UM-ARM-Lab/mps_core/blob/master/mps_msgs/src/mps_msgs/serialization.cpp and replace the repeated usage of functionally identical code with a wrapper.

**Comment by dmcconachie** [comment]
Resolved in #32 . Helper functions added to zlib_helpers.hpp.
